Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -646,7 +646,7 @@ public partial class Translator {
         // translate the body
         TrStmt(m.Body, builder, localVariables, etran);
         m.Outs.Iter(p => CheckDefiniteAssignmentReturn(m.Body.RangeToken.EndToken, p, builder));
-        stmts = builder.Collect(m.Body.Tok);
+        stmts = builder.Collect(m.Body.RangeToken.StartToken); // TODO should this be EndToken?  the parameter name suggests it should be the closing curly
```

Comment:
Which parameter name? Of Collect?

Reply:
Yes, it reads `public StmtList Collect(IToken endCurlyBrace)`

Reply:
Oh I see. This is weird indeed. I think it should be RangeToken.EndToken, but it's true that StartToken probably matches the previous semantics - but these semantics could be wrong.
As a follow-up quick PR, could you experiment replacing StartToken by EndToken? I suspect that it's the token of the "return path", which I would have always thought should be the end curly brace, not the first one which I think is confusing. Perhaps this will be the way to fix it? A low-hanging fruit there.